libs/vkd3d/command.c

```diff
@@ -3324,6 +3333,14 @@ static void vk_access_and_stage_flags_from_d3d12_resource_state(const struct d3d
             case D3D12_RESOURCE_STATE_DEPTH_WRITE:
             case D3D12_RESOURCE_STATE_DEPTH_READ:
                 *stages |= VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
+                *access |= VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT;
+                /* If our default DS layout is attachment optimal, we will not perform implicit
+                 * memory barriers as part of a render pass. */
+                if (d3d12_resource_get_outside_render_pass_depth_stencil_layout(resource) ==
+                        VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL)
+                {
+                    *access |= VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT;
```

Comment:
is this necessary even in the `D3D12_RESOURCE_STATE_DEPTH_READ` case?

Reply:
Given that it's not valid to write to depth in that layout, it's probably fine to omit the access flag in DEPTH_READ.